src/content/docs/2/guide/frontend/nuxt.mdx

```diff
@@ -0,0 +1,103 @@
+---
+title: Nuxt
+---
+
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+Nuxt is a meta-framework for Vue.
+
+## Checklist
+
+- Use SSG by setting `ssr: false`. Tauri doesn't support server based solutions.
+- Set host to `0.0.0.0`.
+- Use `dist/` as `distDir` in `tauri.conf.json`.
+- Compile using `nuxi generate`.
+
+## Example Configuration
+
+<Tabs>
+	<TabItem label="npm">
+	
+```json
+// tauri.conf.json
+{
+	"build": {
+		"beforeDevCommand": "npm run dev",
+		"beforeBuildCommand": "npm run generate",
+		"devPath": "http://localhost:3000",
+		"distDir": "../dist"
+	}
+}
+```
+	
+	</TabItem>
+	<TabItem label="yarn">
+	
+```json
+// tauri.conf.json
+{
+	"build": {
+		"beforeDevCommand": "yarn dev",
+		"beforeBuildCommand": "yarn generate",
+		"devPath": "http://localhost:3000",
+		"distDir": "../dist"
+	}
+}
+```
+	
+	</TabItem>
+	<TabItem label="pnpm">
+	
+```json
+// tauri.conf.json
+{
+	"build": {
+		"beforeDevCommand": "pnpm dev",
+		"beforeBuildCommand": "pnpm generate",
+		"devPath": "http://localhost:3000",
+		"distDir": "../dist"
+	}
+}
+```
+	
+	</TabItem>
+</Tabs>
+
+```ts
+export default defineNuxtConfig({
+	// (optional) Enable the Nuxt devtools
+	devtools: { enabled: true },
+	// Enable SSG
+	ssr: false,
+	// (optional) Disable telemetry
+	telemetry: false,
+	vite: {
+		// Better support for Tauri CLI output
+		clearScreen: false,
+		// Enable environment variables
+		envPrefix: ['VITE_', 'TAURI_'],
+		build: {
+			// (optional) Support modern browsers
+			target: 'es2022',
+			// (optional) Don't minify for debug builds
+			minify: !process.env.TAURI_DEBUG ? 'esbuild' : false,
+			// (optional) Produce sourcemaps for debug builds
+			sourcemap: !!process.env.TAURI_DEBUG,
+		},
+		server: {
+			// Tauri expects a specific port, panic if it can't be bound
+			strictPort: true,
+			// Make sure your mobile device can find you
+			host: '0.0.0.0',
+			hmr: {
+				// Use websocket for mobile hot reloading
+				protocol: 'ws',
+				// Make sure it's available on the network
+				host: '0.0.0.0',
+				// Use a specific port for hmr
+				port: 5183,
+			},
```

Comment:
I couldn't find a lot of details in the vite docs: https://vitejs.dev/config/server-options.html#server-hmr.

- I'm assuming all of these are required for mobile development?
- Does `host` need to be explicitly set or does it inherit it from `server.host`?
- Is there a reason `port` is a different value than the default `3000`?

Reply:
- it's to fix hot reloading for mobile, yes
- in vite it has to be explicit, i **think** it can be skipped for nuxt, unverified
- yes, because it's 2 different servers, 3000 is the http server while 5183 is the websocket server for hmr (dont ask me why that specific port)

Reply:
Cool, will comment on ports in a separate comment.